content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,615 @@
+
+The application in this tutorial _doesn't_ have the ability to send or trade [tokens](issued-currencies.html) or 
+use other [payment types](payment-types.html) like [Escrow](https://xrpl.org/escrow.html) or [Payment Channels](https://xrpl.org/payment-channels.html). However, it provides a foundation 
+that you can implement those and other features on top of.
+
+In addition to the above features, you'll also learn a bit about Events, IPC (inter-process-communication) 
+and asynchronous (async) code in JavaScript.
+
+## Steps
+
+### 0. Project setup
+
+To initialize the project we will create a package.json file with the following content:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/package.json", language="js", lines="1-28") }}
+
+Here we define the libraries our application will use in the `dependencies` section as well as shortcuts for running our application in the `scripts` section. To install those dependencies, run the following command:
+```console
+npm install
+```
+
+This installs the Electron Framework, the xrpl.js client library and a couple of helpers we are going to need for our
+application to work. 
+
+### 1. Hello XRP Ledger
+
+**Full code for this step:** 
+[`1_hello.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/1_hello.js),
+[`view/1_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_preload.js),
+[`view/1_hello.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_hello.html),
+[`view/1_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_renderer.js).
+
+Our first step is to write a "Hello World" like application, that interacts on a very basic level with the XRP Ledger 
+and displays some high-level information about the current ledger state on the screen. Nothing too fancy so far, as we 
+will take care of styling and GUI related coding in a later step:
+
+![Screenshot: Step 1, hello world equivalent](img/javascript-wallet-1.png)
+
+First, we will create an entrypoint for our application, for this we create the file `1_hello.js` with the following content:
+
+`1_hello.js`
+{{ include_code("_code-samples/build-a-wallet/desktop-js/1_hello.js", language="js") }}
+
+This code has two parts: one that creates the application window and one that calls the XRP Ledger API's [ledger method](ledger.html). The code then broadcasts an event with the API response as the payload. The frontend picks up this event and uses the payload to display the index of most recently validated ledger.
+
+To display our results to the user, we need to create the view components that we specified in the `createWindow()` function. For this, we will create a `view` folder and add the following files there:
```

Comment:
If there's ever an action the user should do - try to make it into a numbered list or bullet so it's very clear that you've switched into instruction mode instead of explaining mode. (Otherwise it's easy for them to miss that you were telling them to do something)

```suggestion
In order to make that work, we need to implement the views referenced in `createWindow()` to tell the frontend how to display the payload. 

1. Create a `view` folder next to your `package.lock` file.
2. In the `view` folder, add the following 3 files:
```

Reply:
Obsolete, as the instructions have been completely rewritten